x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1170,21 +1171,44 @@ private Expression simplify(BinaryComparison bc) {
             // optimize only once the expression has a literal on the right side of the binary comparison
             if (bc.right() instanceof Literal) {
                 if (bc.left() instanceof ArithmeticOperation) {
-                    return SimplifyOperation.simplify(bc, typesCompatible);
-                } else if (bc.left() instanceof Neg) {
-                    return reduceNegation(bc);
+                    return simplifyBinaryComparison(bc);
+                }
+                if (bc.left() instanceof Neg) {
+                    return foldNegation(bc);
                 }
             }
             return bc;
         }
 
-        private static Expression reduceNegation(BinaryComparison bc) {
+        private Expression simplifyBinaryComparison(BinaryComparison comparison) {
+            ArithmeticOperation operation = (ArithmeticOperation) comparison.left();
+            // Use symbol comp: SQL operations aren't available in this package (as dependencies)
+            String opSymbol = operation.symbol();
+            // Modulo can't be simplified.
+            if (opSymbol == MOD.symbol()) {
+                return comparison;
+            }
+            OperationSimplifier simplification = null;
+            if (isMulOrDiv(opSymbol)) {
+                simplification = new MulDivSimplifier(comparison);
+            } else if (opSymbol == ADD.symbol() || opSymbol == SUB.symbol()) {
+                simplification = new AddSubSimplifier(comparison);
+            }
+
+            return (simplification == null || simplification.isUnsafe(typesCompatible)) ? comparison : simplification.apply();
+        }
+
+        private static boolean isMulOrDiv(String opSymbol) {
```

Comment:
If this method necessary? I only see it used in one place and the check can check can be directly just life for ADD and SUB.

Reply:
It's being used twice. I'm on the fence tbh - it makes things easier to read in `sign()`, but looks unbalanced in the if-else explicit ADD/SUB check - but would leave it as is now.